server/docs/design/block-persistence.md

```diff
@@ -3,64 +3,235 @@
+7 digits, the last digit is part of the block file name itself, we zip every
+100 block in a single zip )
+
+BlockArchive/.../1/0/2/3/4.zip/0/1023400.blk.zstd -> zipped because we have configed every 100s to be zipped
+BlockLive/.../1/0/2/3/5/0/1023500.blk.zstd -> new block written
+BlockLive/.../1/0/2/3/5/0/1023501.blk.zstd -> new block written
+...
+BlockLive/.../1/0/2/3/5/9/1023599.blk.zstd -> new block written, reaching the 100th mark
+BlockArchive/.../1/0/2/3/5.zip/0/1023500.blk.zstd -> block file ...1023500.blk.zstd to ...1023599.blk.zstd are now zipped
+BlockLive/.../1/0/2/3/6/0/1023600.blk.zstd -> new block
+
+/.../1/0/2/3/5/9/ + 9 (the last digit of the blk file, block number) = the whole block number ...1023599
+```
+
+The algorithm for resolving the path to a `Block` is defined roughly as follows:
 
-**BlockItem** - A BlockItem is the primary data structure passed between the producer, the `hedera-block-node`
-and consumers. The BlockItem description and protobuf definition are maintained in the `hedera-protobuf`
-[project](https://github.com/hashgraph/hedera-protobufs/blob/continue-block-node/documents/api/block/stream/block_item.md).
+1. When writing, initially we only care to write a new `Block` as a file to the
+   local filesystem in the correct place, based on the resolved path from the
+   `BlockNumber` utilizing the `trie` structure.
+1. When writing a new `Block` as file, we need to check if the path to the file
+   we want to write already exists, if it does, there is some problem, we should
+   handle this case directly in our service.
+1. When reading, we should be able to resolve the path to a `Block` as file that
+   we want to read, initially attempting to read the file before it would be
+   zipped, if not found, we need to search it in the respective zip, else if not
+   found, then we have a problem, we need to handle that case inside our
+   service.
+1. A separate process/thread will periodically go through our `trie` structure
+   and will be zipping the `Block` files as per configured amounts as shown
+   visually above.
 
-**Block** - A Block is the base element of the block stream at rest. At present, it consists of an ordered collection of
-BlockItems. The Block description and protobuf definition are maintained in the `hedera-protobuf`
-[project](https://github.com/hashgraph/hedera-protobufs/blob/continue-block-node/documents/api/block/stream/block.md).
+#### Specific implementations
+
+The specific implementations of the defined [abstractions](#abstractions) as
+listed above are:
+
+1. `BlockWriter` - `com.hedera.block.server.persistence.storage.write.BlockAsLocalFileWriter`
+1. `BlockReader` - `com.hedera.block.server.persistence.storage.read.BlockAsLocalFileReader`
+1. `BlockRemover` - `com.hedera.block.server.persistence.storage.remove.BlockAsLocalFileRemover`
+1. `BlockPathResolver` - `com.hedera.block.server.persistence.storage.path.BlockAsLocalFilePathResolver`
```

Comment:
Including specific concrete classes seems overly detailed for a high level design.
I recall mentioning this in the first version of this document, and it becomes even more clear with this update.
Simply removing this section (for each option) would, I believe, make the design clearer and significantly reduce the need to update high-level design every time we change details best understood from detail design documents.

Reply:
✅ Done.